### PR TITLE
task8 - nest app cart api

### PR DESCRIPTION
**Pull request by code-by**
task8 - nestJS cart api

**Assignment:** 
https://github.com/rolling-scopes-school/aws/blob/main/aws-developer/08_integration_with_sql_database/task.md

**Points evaluation**: 45

**What is done:**
1. Nest App updated and deployed as wrapped AWS Lambda
https://74df9jyrqg.execute-api.eu-west-1.amazonaws.com/cart
for test purpose to obtain `200` status:
https://74df9jyrqg.execute-api.eu-west-1.amazonaws.com/cart?test

2. RDS tables `carts`, ` cart_items` created, SQL script with examples implemented (`/scripts` folder)
rss_shop - public

Unfortunately I have spent much time to find out what was wrong with my nest/cdk/node and why deployment lambda unable to find handler, and now not all tasks implemented. I hope I could done with part of tasks before deadline, at least task 8.2.
You could check as is, or, if this possible, review before 4 hrs before deadline. Thank you.

### CODE REVIEW

File: cdk/lib/nestAppStack.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    // lambdas
+    const nestAppLambda = new lambdaNode.NodejsFunction(
+      this,
+      "nestAppLambdaNew",
+      {
+        functionName: "nestAppLambdaNew",
+        entry: path.join(__dirname, "../../dist/main.js"),
+        memorySize: 1024,
+        timeout: cdk.Duration.seconds(5),
```
Comment (takegu) [inline cdk/lib/nestAppStack.ts]: You should build your app with comment `npx nest build --webpack`.

File: cdk/lib/nestAppStack.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    );
+
+    // gateway
+    const lambdatIntegration = new HttpLambdaIntegration('CartLambda', nestAppLambda);
+
+    const cartHTTPApi = new HttpApi(this, 'CartHTTPApi', {
+    });
+    
+    cartHTTPApi.addRoutes({
+      path: '/cart',
```
Comment (takegu) [inline cdk/lib/nestAppStack.ts]: Try to use for path: `path: '/{proxy+}',`

Comment (takegu) [reply cdk/lib/nestAppStack.ts]: maybe it will be helpful

File: cdk/lib/nestAppStack.ts (hunk 1 above)
Comment (code-by) [reply cdk/lib/nestAppStack.ts]: thank you for advice, it is build into single file, but aws lambda unable to find handler (it find it if I manually modify my index.js and change export to `module.exports`.
the only solution that works for me is building w/o webpack and its works
can't figure out why it not working when compiling with webpack

**Pull request merged**